src/sp/it/pl/layout/widget/WidgetManager.kt

```diff
@@ -211,9 +211,13 @@ class WidgetManager(private val windowManager: WindowManager, private val userEr
 
         fun findSrcFiles() = widgetDir.seqChildren().filter { it.hasExtension("java", "kt") }
 
+        fun findClassFile() = widgetDir.childOf("$widgetName.class")
+
+        fun findClassFiles() = widgetDir.seqChildren().filter { it hasExtension "class" }
+
         fun computeClassPath(): String = computeClassPathElements().joinToString(classpathSeparator)
 
-        private fun computeClassPathElements() = sequenceOf(".")+getAppJarFile()+(findAppLibFiles()+findLibFiles()).map { it.absolutePath }
+        private fun computeClassPathElements() = sequenceOf(".")+getAppJarFile()+widgetDir+(findAppLibFiles()+findLibFiles()).map { it.absolutePath }
```

Comment:
why does it start with a dot?

Reply:
I no longer remember, but it is necessary. On the other hand the widgetDir was only added on a whim and should be removed.

Note mixed Kotlin-Java widgets still do not compile successfully. However java files in a widget directory no longer cause compilation errors - so long as they are not used. This is almost useless feature, if it wasnt for the fact that I have created such situation here.